src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Match.SRM.cs

```diff
@@ -1,63 +1,32 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
-using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions.SRM;
 
 namespace System.Text.RegularExpressions
 {
     public partial class Regex
     {
-        /// <summary>Used iff <see cref="_srm"/> is non-null.</summary>
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        internal Match? RunSRM(bool quick, string input, int beg, int startat, int length, int prevlen)
-        {
-            Debug.Assert(_srm != null);
-
-            int endAt = beg + length;
-
-            // If the previous match was empty, advance by one before matching
-            // or terminate the matching if there is no remaining input to search in
-            if (prevlen == 0)
-            {
-                if (startat == endAt)
-                    return RegularExpressions.Match.Empty;
-
-                startat += 1;
-            }
+        // TODO: Avoid rooting Serialize/Deserialize in ILLink purely for use by tests
+        // TODO: Only compile SaveDGML into Debug build
+        // TODO: Figure out what to do about Serialize/Deserialize in general (e.g. delete them, expose them, use them in source generated code, etc.)
 
-            SRM.Match? match = _srm._matcher.FindMatch(quick, input, startat, endAt);
-            if (quick)
-            {
-                if (match is null)
-                {
-                    return null;
-                }
-            }
-            else
+        internal string Serialize()
+        {
+            if (factory is not SymbolicRegexRunnerFactory srmFactory)
             {
-                Debug.Assert(match is not null);
-                if (match.Success)
-                {
-                    var m = new Match(this, 1, input, beg, length, startat);
-                    m._matches[0][0] = match.Index;
-                    m._matches[0][1] = match.Length;
-                    m._matchcount[0] = 1;
-                    m.Tidy(match.Index + match.Length);
-                    return m;
-                }
+                throw new NotSupportedException();
```

Comment:
for whatever reason, we use PlatformNotSupportedException for the existing serialize API. NSE does seem more correct but it's not consistent?

Reply:
From my perspective,  this whole function is just a placeholder. It's currently not used anywhere, and assuming it's worth it's weight,  I think our goal should be for it to be an implementation detail as part of source generating non-backtracking regexes.